node.go

```diff
@@ -205,7 +205,10 @@ func (node *Node) _hash() []byte {
 	if err := node.writeHashBytes(buf); err != nil {
 		panic(err)
 	}
-	h.Write(buf.Bytes())
+	_, err := h.Write(buf.Bytes())
+	if err != nil {
```

Comment:
we weren't checking before so there is a uber low chance of erroring, so panic is fine here